app/src/main/java/com/tomtom/sdk/examples/maps/ConfigurableMapActivity.kt

```diff
@@ -167,9 +200,142 @@ class ConfigurableMapActivity : AppCompatActivity() {
+            } else {
+                if(mapPreferencesToUpgrade.containsKey(title)) {
+                    mapPreferencesToUpgrade.remove(title)
+                }
+                false
+            }
+        }
+    }
+
+    private fun upgradeMap() {
+        val onStyleLoadedCallback = object: StyleLoadingCallback {
+            override fun onSuccess() {
+                currentMapStyleDescriptor = mapStyleDescriptorToUpgrade
+            }
+
+            override fun onFailure(failure: LoadingStyleFailure) {
+                Log.d("ConfigurableMapActivity.kt: Could not upgrade map with style descriptor", failure.message)
+            }
+        }
+        tomTomMap.setStyleMode(mapBaseStyleToUpgrade)
+        tomTomMap.loadStyle(mapStyleDescriptorToUpgrade, onStyleLoadedCallback)
+        currentBaseMapStyle = mapBaseStyleToUpgrade
+    }
+
+    private fun upgradeMapPreferences() {
+        mapPreferencesToUpgrade.forEach { (key, value) ->
+            when(value) {
+                MapPreference.SHOW_TRAFFIC_FLOW -> tomTomMap.showTrafficFlow()
+                MapPreference.HIDE_TRAFFIC_FLOW -> tomTomMap.hideTrafficFlow()
+                MapPreference.SHOW_HILL_SHADING -> tomTomMap.hideHillShading()
+                MapPreference.HIDE_HILL_SHADING -> tomTomMap.hideHillShading()
+                MapPreference.SHOW_VEHICLE_RESTRICTIONS -> tomTomMap.showVehicleRestrictions()
+                MapPreference.HIDE_VEHICLE_RESTRICTIONS -> tomTomMap.hideVehicleRestrictions()
+            }
+            currentMapPreferences[key] = value
+        }
+
+        mapPreferencesToUpgrade.clear()
+    }
+
+    private fun setUpRecyclerListForBaseMapStyle(baseMapStyleRecyclerList: RecyclerView) {
+        createGridLayout(baseMapStyleRecyclerList, LAYOUT_SPAN_2_RECYCLER_VIEW)
+        createAdapterForBaseMapStyleData(baseMapStyleRecyclerList)
+    }
+
+    private fun setUpRecyclerListForMapPreferences(mapPreferenceRecyclerList: RecyclerView) {
+        createGridLayout(mapPreferenceRecyclerList, LAYOUT_SPAN_1_RECYCLER_VIEW)
+        createAdapterForMapPreferencesData(mapPreferenceRecyclerList)
+    }
+
+    private fun createGridLayout(recyclerList: RecyclerView, layoutSpanCount: Int) {
+        val gridLayoutManager = GridLayoutManager(this, layoutSpanCount)
+        recyclerList.layoutManager = gridLayoutManager
+    }
+
+    private fun createAdapterForBaseMapStyleData(baseMapStyleRecyclerList: RecyclerView) {
+        val baseMapStyleDataAdapter = BaseMapStyleDataAdapter(baseMapStyleRecyclerList, baseMapStyleList!!, listener, currentBaseMapStyle, currentMapStyleDescriptor)
+        baseMapStyleRecyclerList.adapter = baseMapStyleDataAdapter
+    }
+
+    private fun createAdapterForMapPreferencesData(mapPreferenceRecyclerList: RecyclerView) {
+        val mapPreferenceDataAdapter = MapPreferenceDataAdapter(mapPreferencesList!!, listener, currentMapPreferences)
+        mapPreferenceRecyclerList.adapter = mapPreferenceDataAdapter
+    }
```

Comment:
I don't see any advantage of extracting this methods, they are called in already small specific methods